### PR TITLE
Fix continueOnError when using latest JDK

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes an issue where `continueOnError` was incorrectly being set to `true` allowing test runs to pass with failed tests.

### CODE REVIEW

File: sdk/core/azure-core/src/test/java/com/azure/core/util/FluxUtilTest.java [hunk 1]
```diff
@@ -236,17 +236,18 @@ public void testWriteWritableChannel() throws Exception {
...
-                c.close();
-            } catch (IOException ex) {
-                throw new UncheckedIOException(ex);
-            }
-        });
-
-        StepVerifier.create(writeFile)
-            .expectError(expectedException)
-            .verify(Duration.ofSeconds(30));
+        throw new RuntimeException("Test");
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/test/java/com/azure/core/util/FluxUtilTest.java:239]: Used for testing the fix, will be removed once validated.

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 2]
```diff
@@ -95,3 +95,4 @@ jobs:
           TestOptions: ${{ parameters.TestOptions }}
           TestParallelization: ${{ parameters.TestParallelization }}
           BuildParallelization: ${{ parameters.BuildParallelization }}
+          ContinueLatestBuildOnError: ${{ eq(variables['LatestJdkVersion'], variables['JavaTestVersion']) }}
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.tests.yml:98]: I don't think this syntax is correct as those variables aren't available at template compile type so they will always be empty.

File: eng/pipelines/templates/steps/build-and-test.yml [hunk 3]
```diff
@@ -54,7 +57,7 @@ steps:
     env:
       AZURE_VERSION_OVERRIDE_TESTS: ${{ parameters.TestVersionSupport }}
     condition: and(succeeded(), ne(variables['TestFromSource'], 'true'))
-    continueOnError: ${{ eq(variables['LatestJdkVersion'], variables['JavaTestVersion']) }}
+    continueOnError: ${{ parameters.ContinueLatestBuildOnError }}
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-and-test.yml:60]:
```suggestion
    continueOnError: eq(variables['LatestJdkVersion'], variables['JavaTestVersion'])
```
Does that syntax work correctly?

Comment (alzimmermsft) [reply eng/pipelines/templates/steps/build-and-test.yml:60]: This results in a syntax error

image

**Comment by alzimmermsft** [comment]
Closing as this isn't the right path to fix this, check out #31899

**Pull request closed without merging**